src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -69,134 +75,109 @@ public bool TryUnparse(StringBuilder sb)
                 return TryUnparseCore(sb);
             }
         }
-
-        private static class Defaults
-        {
-            public const bool Bag = false;
-        }
-
         public sealed class Arguments
         {
             [Argument(ArgumentType.Multiple, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
             public Column[] Column;
 
             [Argument(ArgumentType.AtMostOnce,
                 HelpText = "Whether to combine multiple indicator vectors into a single bag vector instead of concatenating them. This is only relevant when the input is a vector.")]
-            public bool Bag = Defaults.Bag;
+            public bool Bag = KeyToVectorEstimator.Defaults.Bag;
         }
 
-        internal const string Summary = "Converts a key column to an indicator vector.";
+        public class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+            public readonly bool Bag;
+            public ColumnInfo(string input, string output, bool bag = KeyToVectorEstimator.Defaults.Bag)
+            {
+                Input = input;
+                Output = output;
+                Bag = bag;
+            }
+        }
 
-        public const string LoaderSignature = "KeyToVectorTransform";
-        private static VersionInfo GetVersionInfo()
+        internal sealed class ColInfo
         {
-            return new VersionInfo(
-                modelSignature: "KEY2VECT",
-                verWrittenCur: 0x00010001, // Initial
-                verReadableCur: 0x00010001,
-                verWeCanReadBack: 0x00010001,
-                loaderSignature: LoaderSignature);
+            public readonly string Name;
+            public readonly string Source;
+            public readonly ColumnType TypeSrc;
+
+            public ColInfo(string name, string source, ColumnType type)
+            {
+                Name = name;
+                Source = source;
+                TypeSrc = type;
+            }
         }
 
         private const string RegistrationName = "KeyToVector";
+        private readonly bool[] _bags;
+        private readonly int[] _valueCounts;
+        private readonly int[] _sizes;
 
-        // These arrays are parallel to Infos.
-        // * _bag indicates whether vector inputs should have their output indicator vectors added
-        //   (instead of concatenated). This is faithful to what the user specified in the Arguments
```

Comment:
why remove the comment?

Reply:
I left comment for bags, everything is no longer  exist in code

---
In reply to: [216121559](https://github.com/dotnet/machinelearning/pull/858#discussion_r216121559) [](ancestors = 216121559)